components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -598,6 +599,25 @@ void ConversationDriver::OnSuggestedQuestionsResponse(
   DVLOG(2) << "Got questions:" << base::JoinString(suggestions_, "\n");
 }
 
+bool ConversationDriver::ShouldUnlinkPageContent() {
+  // Only unlink if panel is closed and there is no conversation history.
+  // When panel is open or has existing conversation, do not change the state.
+  return !is_conversation_active_ && chat_history_.empty();
+}
+
+void ConversationDriver::SummarizeSelectedText(
+    const std::string& selected_text,
+    bool should_unlink_page_content) {
+  if (should_unlink_page_content) {
+    SetShouldSendPageContents(false);
+  }
```

Comment:
Having this as a param seems inconsistent when the other similar public functions don't. I understand that you've added `ShouldUnlinkPageContent` to make sure that page content isn't linked mysteriously from "external" requests. At the moment these requests are from 1) context menu and 2) omnibox.

Omnibox is calling `SubmitHumanConversationEntry` which doesn't have the `should_unlink_page_content` param, and context menu is calling `SummarizeSelectedText` which does.

Perhaps we can make this a little more readable and consistent. How about we contain this logic in the same place? Perhaps ConversationDriver can simply unlink the page content if the conditions in `ShouldUnlinkPageContent` are `true` and we can make `ShouldUnlinkPageContent` a private member?

In other words, `SubmitHumanConversationEntry` could have the following inserted at the beginning:

```cpp
if (ShouldUnlinkPageContent()) {
  SetShouldSendPageContents(false);
}
```
or do this logic inline, or if it has to happen in multiple places, change the name of `ShouldUnlinkPageContents` to `MaybeUnlinkPageContents` and it can perform the decision.

Reply:
fixed by https://github.com/brave/brave-core/pull/21670/commits/fc4a6dd8f6dd5a95d867e959c1a17d46a19b8712
Can't put inside `SubmitHumanConversationEntry` as the panel activation status already changed when it's called.